rules/msk_topic_config.go

```diff
@@ -305,48 +305,179 @@ func (r *MSKTopicConfigRule) validateCleanupPolicy(
 
 const (
 	retentionTimeAttr = "retention.ms"
+	millisInOneDay    = 1 * 24 * 60 * 60 * 1000
+	// The threshold on retention time when remote storage is supported.
+	tieredStorageThresholdInDays    = 3
+	tieredStorageEnableAttr         = "remote.storage.enable"
+	tieredStorageEnabledValue       = "true"
+	localRetentionTimeAttr          = "local.retention.ms"
+	localRetentionTimeInDaysDefault = 1
 )
 
 /*	Putting an invalid value by default to force users to put a valid value */
-var retentionTimeDefTemplate = fmt.Sprintf(`"%s" = "???"`, retentionTimeAttr)
+var (
+	retentionTimeDefTemplate = fmt.Sprintf(`"%s" = "???"`, retentionTimeAttr)
+	enableTieredStorage      = fmt.Sprintf(`"%s" = "%s"`, tieredStorageEnableAttr, tieredStorageEnabledValue)
+	localRetentionTimeFix    = fmt.Sprintf(
+		`# keep data in hot storage for %d day
+     "%s" = "%d"`,
+		localRetentionTimeInDaysDefault,
+		localRetentionTimeAttr,
+		localRetentionTimeInDaysDefault*millisInOneDay)
+)
 
 func (r *MSKTopicConfigRule) validateRetentionForDeletePolicy(
 	runner tflint.Runner,
 	config *hclext.Attribute,
 	configKeyToPairMap map[string]hcl.KeyValuePair,
 ) error {
-	rtPair, hasRt := configKeyToPairMap[retentionTimeAttr]
-	if !hasRt {
-		msg := fmt.Sprintf("%s must be defined on a topic with cleanup policy delete", retentionTimeAttr)
+	retentionTime, err := r.getAndValidateRetentionTime(runner, config, configKeyToPairMap)
+	if err != nil || retentionTime == nil {
+		return err
+	}
+
+	if *retentionTime <= tieredStorageThresholdInDays*millisInOneDay && !isInfiniteRetention(*retentionTime) {
+		return nil
+	}
```

Comment:
nit/style: could be more explicit in separating error handling from the logic flow (the check could also possibly be bundled with the following `if` to put all the logic in one place: if `retentionTime == nil || ( ... )`)

```suggestion
	if err != nil {
		return err
	}

	if retentionTime == nil {
		return nil
	}
	if *retentionTime <= tieredStorageThresholdInDays*millisInOneDay && !isInfiniteRetention(*retentionTime) {
		return nil
	}
```

Reply:
Yea.. I had my doubts about this one too when writing it.. I'll split it

Reply:
Fixed in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/22/commits/a6c1fd9dccb0507d7e4f60ca4ea3e04a97cf899e